keylime/cloud_verifier_tornado.py

```diff
@@ -1005,21 +1071,27 @@ async def process_agent(agent, new_operational_state, failure=Failure(Component.
         except SQLAlchemyError as e:
             logger.error("SQLAlchemy Error: %s", e)
 
+        # Load agent's IMA policy
+        try:
+            ima_policy = session.query(VerfierMain).options(joinedload(VerfierMain.ima_policy)).filter_by(agent_id=str(agent["agent_id"])).first().ima_policy
```

Comment:
This isn't the most efficient way to query the `ima_policy` table. It's basically querying the `verifiermain` table and joining it to the `ima_policy` table and then you discard everything but the allowlist. It's more efficient to query the `ima_policy` table directly filtering on the `agent_id`.

Reply:
But if you already have the agent object loaded from the DB and you used `joinedload()` when you retrieved it, then it should already have the `ima_policy` available as an attribute on the `agent` object.

Reply:
I've removed this query in favor of piggybacking on the existing query for `stored_agent` a bit further up. Should be more efficient.